src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ColorSchemesPageViewModel.h"
+#include "ColorSchemesPageViewModel.g.cpp"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    ColorSchemesPageViewModel::ColorSchemesPageViewModel(const Model::CascadiaSettings& settings) :
+        _settings{ settings }
+    {
+        _MakeColorSchemeVMsHelper();
+        CurrentScheme(_AllColorSchemes.GetAt(0));
+    }
+
+    void ColorSchemesPageViewModel::UpdateSettings(const Model::CascadiaSettings& settings)
+    {
+        _settings = settings;
+        _MakeColorSchemeVMsHelper();
+    }
+
+    void ColorSchemesPageViewModel::_MakeColorSchemeVMsHelper()
+    {
+        Windows::Foundation::Collections::IObservableVector<Editor::ColorSchemeViewModel> allColorSchemes{ single_threaded_observable_vector<Editor::ColorSchemeViewModel>() };
+        const auto& colorSchemeMap{ _settings.GlobalSettings().ColorSchemes() };
+        for (const auto& pair : colorSchemeMap)
+        {
+            const auto viewModel = Editor::ColorSchemeViewModel(pair.Value());
+            allColorSchemes.Append(viewModel);
+        }
+        _AllColorSchemes = allColorSchemes;
+    }
+
+    void ColorSchemesPageViewModel::RequestSetCurrentScheme(Editor::ColorSchemeViewModel scheme)
+    {
+        CurrentScheme(scheme);
+
+        // Update the state of the page
+        _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"CanDeleteCurrentScheme" });
+    }
+
+    bool ColorSchemesPageViewModel::RequestRenameCurrentScheme(winrt::hstring newName)
+    {
+        // check if different name is already in use
+        const auto oldName{ CurrentScheme().Name() };
+        if (newName != oldName && _settings.GlobalSettings().ColorSchemes().HasKey(newName))

```

Comment:
So this was code just moved over from `ColorSchemes.cpp`, but I believe we have it this way so that if the user wants to 'abandon' their rename by just re-entering the existing name, we don't throw an error at them